gio/src/subclass/file_enumerator.rs

```diff
@@ -0,0 +1,395 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::{prelude::*, subclass::prelude::*, translate::*, Error};
+
+use crate::{ffi, traits::FileEnumeratorExt, Cancellable, FileEnumerator, FileInfo, IOErrorEnum};
+
+// Support custom implementation of virtual functions defined in `gio::ffi::GFileEnumeratorClass` except pairs `xxx_async/xxx_finish` for which GIO provides a default implementation.
```

Comment:
Overriding these default impls might still be useful for subclasses (if they can do something better than blocking IO). That can be added in another PR but the comment sounds right now like it's not useful.